src/test/java/seedu/address/model/person/PatientNameContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,106 @@
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        PatientNameContainsKeywordsPredicate firstPredicateCopy =
+                new PatientNameContainsKeywordsPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different person -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    public void test_predicateReturnsPatientsOnly() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("Alice");
+        PatientNameContainsKeywordsPredicate firstPredicate =
+                new PatientNameContainsKeywordsPredicate(firstPredicateKeywordList);
+
+        // Object type Patient -> returns true
+        Patient p = new Patient(new Nric("T1234567A"), new Name("Alice"), new DoB("2001-01-01"), new Phone("98765432"));
+        assertTrue(firstPredicate.test(p));
+
+        Doctor d = new Doctor(new Nric("T1234567A"), new Name("Alice"), new DoB("2001-01-01"), new Phone("98765432"));
+        // Object type Doctor -> returns false
+        assertFalse(firstPredicate.test(d));
+    }
+
+    @Test
+    public void test_nameContainsKeywords_returnsTrue() {
+        // One keyword
+        PatientNameContainsKeywordsPredicate predicate =
+                new PatientNameContainsKeywordsPredicate(Collections.singletonList("Alice"));
+        assertTrue(predicate.test(new PatientBuilder().withName("Alice Bob").build()));
+
+        // Multiple keywords
+        predicate = new PatientNameContainsKeywordsPredicate(Arrays.asList("Alice", "Bob"));
+        assertTrue(predicate.test(new PatientBuilder().withName("Alice Bob").build()));
+
+        // Only one matching keyword
+        predicate = new PatientNameContainsKeywordsPredicate(Arrays.asList("Bob", "Carol"));
+        assertTrue(predicate.test(new PatientBuilder().withName("Alice Carol").build()));
+
+        // Mixed-case keywords
+        predicate = new PatientNameContainsKeywordsPredicate(Arrays.asList("aLIce", "bOB"));
+        assertTrue(predicate.test(new PatientBuilder().withName("Alice Bob").build()));
+    }
+
+    @Test
+    public void test_nameDoesNotContainKeywords_returnsFalse() {
+        // Zero keywords
+        PatientNameContainsKeywordsPredicate predicate =
+                new PatientNameContainsKeywordsPredicate(Collections.emptyList());
+        assertFalse(predicate.test(new PatientBuilder().withName("Alice").build()));
+
+        // Non-matching keyword
+        predicate = new PatientNameContainsKeywordsPredicate(Arrays.asList("Carol"));
+        assertFalse(predicate.test(new PatientBuilder().withName("Alice Bob").build()));
+
+        // // Keywords match phone, email and address, but does not match name
```

Comment:
I think this is fine to leave out for now since it isn't testable - this requires searching by other fields - search by phone/email/address which we dont support currently